src/main/java/net/coderbot/iris/pipeline/newshader/TriforcePatcher.java

```diff
@@ -193,6 +194,37 @@ public static String patch(String source, ShaderType type, AlphaTest alpha, bool
 		ProgramBuilder.MACRO_CONSTANTS.getDefineStrings().forEach(defineString ->
 				transformations.injectLine(Transformations.InjectionPoint.DEFINES, defineString + "\n"));
 
+		// Inject the legacy lines transformation after everything else
+		if(type == ShaderType.VERTEX && injectLegacyLines) {
+			transformations.injectLine(Transformations.InjectionPoint.BEFORE_CODE, "uniform float iris_LineWidth;");
+			// TODO: Need a better way to do this
+			transformations.replaceExact("uniform vec3 cameraPosition;", ""); // Required if shader has cameraPosition inside a guard
+			transformations.injectLine(Transformations.InjectionPoint.BEFORE_CODE, "uniform vec3 cameraPosition;");
+
+			transformations.replaceExact("uniform float viewWidth;", "");
+			transformations.replaceExact("uniform float viewHeight;", "");
+			transformations.replaceExact("uniform float viewWidth, viewHeight;", ""); // This is sooo scuffed
+			transformations.injectLine(Transformations.InjectionPoint.BEFORE_CODE, "uniform float viewWidth;");
+			transformations.injectLine(Transformations.InjectionPoint.BEFORE_CODE, "uniform float viewHeight;");
+
+			transformations.injectLine(Transformations.InjectionPoint.BEFORE_CODE, "void _iris_internal_legacy_lines() {\n" +
+					"vec4 _iris_internal_ndcStart4 = iris_ProjMat * iris_ModelViewMat * vec4(Position, 1.0);\n" +
+					"vec4 _iris_internal_ndcEnd4 = iris_ProjMat * iris_ModelViewMat * vec4(Position + Normal, 1.0);\n" +
+					"vec3 _iris_internal_ndcStart = _iris_internal_ndcStart4.xyz / _iris_internal_ndcStart4.w;\n" +
+					"vec3 _iris_internal_ndcEnd = _iris_internal_ndcEnd4.xyz / _iris_internal_ndcEnd4.w;\n" +
+					"vec2 _iris_internal_screenSize = vec2(viewWidth, viewHeight);\n" +
+					"vec2 _iris_internal_lineDir = normalize(_iris_internal_ndcEnd.xy - _iris_internal_ndcStart.xy);\n" +
+					"vec2 _iris_internal_lineOffset = (vec2(-_iris_internal_lineDir.y, _iris_internal_lineDir.x) * (iris_LineWidth / _iris_internal_screenSize));\n" +
+					"if(_iris_internal_lineOffset.x < 0) { _iris_internal_lineOffset = _iris_internal_lineOffset * -1; }\n" +
+					"if(gl_VertexID % 2 == 0) {\n" +
+					"gl_Position = (vec4(_iris_internal_ndcStart, 1.0) + vec4(_iris_internal_lineOffset, 0.0, 0.0) - vec4(0.0, 0.0, 0.0001, 0.0)) * _iris_internal_ndcStart4.w;\n" +
+					"} else {\n"+
+					"gl_Position = (vec4(_iris_internal_ndcStart, 1.0) - vec4(_iris_internal_lineOffset, 0.0, 0.0) - vec4(0.0, 0.0, 0.0001, 0.0)) * _iris_internal_ndcStart4.w;\n" +
```

Comment:
Is there a reason for this duplicate code?

Reply:
Its not duplicate. Every other vertex must be pushed in the opposite direction. Otherwise the line would not have any width.

Reply:
Oh, didn't see the difference, my bad.